resolve.c

```diff
@@ -85,11 +85,19 @@ void resolveClients(bool onlynew)
 		if(onlynew && !clients[clientID].new)
 			continue;
 
+		// Lock data when obtaining IP of this client
 		lock_shm();
+		const char* ipaddr = getstr(clients[clientID].ippos);
+		unlock_shm();
 
-		clients[clientID].namepos = addstr(resolveHostname(getstr(clients[clientID].ippos)));
```

Comment:
Doing all this in one line was the origin of the observed issues with name resolution as `resolveHostname()` cannot run when the lock is still obtained. Hence, I have split this one-liner into three blocks where only the first and the third are run locked. This is fine as the function `resolveHostname()` does neither access not modify FTL's datastructure and, hence, does not need to run locked (actually, it *cannot* even run locked).